lib/engines/csvEngine.js

```diff
@@ -0,0 +1,163 @@
+var _ = require("lodash");
+var Bromise = require("bluebird");
+var converter = require("json-2-csv");
+var mqtt = require("mqtt");
+var fs = require('fs');
+
+var MQTT_PORT = "1883";
+
+/***
+Config:
+   "generation": {
+        "mode": "csvEngine",
+        "csvEngine": {
+            "fileRepository": ".",
+            "csvOptions": "",
+            "timestamp": 1,
+            "value": 2
+            }
+        }
+***/
+
+function CSVEngine(simulator) {
+
+    this.config = simulator.simulation.generation.csvEngine;
+    this.simulator = simulator;
+    this.mqttServer = "tcp://" + simulator.configuration.server;
+    this.dataGenerators = {};
+}
+
+CSVEngine.prototype.generateValues = function(dataDescriptor, system) {
+    var self = this;
+    var result = {};
+
+    var filename = this.config.fileRepository + "/" + system.name + ".csv";
+    console.log("    Reading data [" + filename + "]");
+    var config = this.config;
+    return new Bromise(function(resolve, reject) {
+        fs.readFile(filename, 'utf8', function(err, data) {
+            if (err) {
+                reject(err);
+            }
+            converter.csv2json(data, function(err, json) {
+                if (err) {
+                    console.log(err);
+                    reject(err);
+                }
+
+                try {
+                    // Browse each data
+                    console.log("    Generating", json.length, "data");
+                    json.forEach(function(object) {
+                        var timestamp = object[self.getTimestampConfig(dataDescriptor)] + "000";
+                        result[timestamp] = {};
+                        result[timestamp][dataDescriptor] = object[self.getDataValueConfig(dataDescriptor)];
+                    });
+                    resolve(result);
+                } catch (e) {
+                    console.error("# Data gen error:", e);
+                    reject(e);
+                }
+            }, config.csvOptions);
+        });
+    });
+};
+
+CSVEngine.prototype.generateData = function(system) {
+    var self = this;
+    console.log("Generate data for system:", system.name);
+    //var datas = {};
+    try {
+        return _.reduce(self.getGenerationData(), function(datas, dataDescriptor, dataName) {
+            console.log("generate value for data:", dataName);
+
+            return self.generateValues(dataName, system).then(function(data) {
+                return _.extend(datas, data);
```

Comment:
Again arrow functions :muscle: 

```js
try {
  return _.reduce(self.getGenerationData(), (datas, dataDescriptor, dataName) => {
      console.log("generate value for data:", dataName);
      // No need for paranthesis when there's a single param
      // like here -------------------------------------,
      return self.generateValues(dataName, system).then(data => _.extend(datas, data));
      // And no need to wrap with `{ }` and use `return`            ^
      // When there's a single instruction like here  --------------'
  });
}
```

Reply:
I replaced this code by your sample code (no change):
return _.reduce(self.getGenerationData(), function(datas, dataDescriptor, dataName) {
            console.log("generate value for data:", dataName);

            return self.generateValues(dataName, system).then(function(data) {
                return _.extend(datas, data);
            });
        }, {});